rake_lib/repo.rb

```diff
@@ -123,8 +123,10 @@ def update_branches
           valid_branches << line.split('/').last.strip
         end
       end
-
+      # Adding branches in format x.x.x as per new release guidelines
+      @documentation_publish_branches = valid_branches.select{ |branch| branch[/^\d*\.\d*.\d*$/] }
       @branches = valid_branches.uniq.sort
+      puts "Branches for which documentation will be published: #{documentation_publish_branches.join(', ')}"
```

Comment:
this is bad ruby. you've changed the return value of this method to nil. it use to be `@branches`

also the puts should be using the branches attr, not the unsorted non-deduped documentation_publish_branches